scripts/infinite-zoom.py

```diff
@@ -12,38 +12,55 @@
 
 from iz_helpers import shrink_and_paste_on_blank, write_video
 from webui import wrap_gradio_gpu_call
-from modules import script_callbacks
+from modules import script_callbacks, scripts
 import modules.shared as shared
 from modules.processing import (
     process_images,
     StableDiffusionProcessingTxt2Img,
     StableDiffusionProcessingImg2Img,
 )
 
+from scripts import postprocessing_upscale
+
 from modules.ui import create_output_panel, plaintext_to_html
+import modules.sd_models
+import modules.sd_samplers
+
+from modules import scripts
+
+usefulDirs = scripts.basedir().split(os.sep)[
+    -2:
+]  # contains install and our extension foldername
+jsonprompt_schemafile = (
+    usefulDirs[0] + "/" + usefulDirs[1] + "/scripts/promptschema.json"
+)
 
-available_samplers = [
-    "DDIM",
-    "Euler a",
-    "Euler",
-    "LMS",
-    "Heun",
-    "DPM2",
-    "DPM2 a",
-    "DPM++ 2S a",
-    "DPM++ 2M",
-    "DPM++ SDE",
-    "DPM fast",
-    "DPM adaptive",
-    "LMS Karras",
-    "DPM2 Karras",
-    "DPM2 a Karras",
-    "DPM++ 2S a Karras",
-    "DPM++ 2M Karras",
-    "DPM++ SDE Karras",
-]
-default_prompt = "A psychedelic jungle with trees that have glowing, fractal-like patterns, Simon stalenhag poster 1920s style, street level view, hyper futuristic, 8k resolution, hyper realistic"
-default_negative_prompt = "frames, borderline, text, character, duplicate, error, out of frame, watermark, low quality, ugly, deformed, blur"
+available_samplers = [s.name for s in modules.sd_samplers.samplers]
```

Comment:
TODO: exclude only txt2img samplers like UniPC

Reply:
we provide two settings now: model for txt2img and model for img2img. So in one list we see all model, and in the other a filtered one

Reply:
like this
available_samplers = [s.name for s in modules.sd_samplers.samplers if "UniPc" not in s.name]